### PR TITLE
Implement protocol renaming feature post-extraction

### PR DESCRIPTION
**Pull request by ilkilic**
This PR introduces a feature allowing users to provide a protocols_mapping dictionary. This functionality enables the renaming of protocol identifiers post-extraction based on user-defined mappings.

### CODE REVIEW

File: bluepyemodel/evaluation/fitness_calculator_configuration.py [hunk 1]
```diff
@@ -332,15 +339,20 @@ def init_from_bluepyefe(self, efeatures, protocols, currents, threshold_efeature
         self.protocols = []
         self.efeatures = []
 
+        if protocols_mapping:
+            self.validation_protocols = [
+                protocols_mapping.get(vp, vp) for vp in self.validation_protocols
```
[the remark this answers is not in the document]
Comment (ilkilic) [reply bluepyemodel/evaluation/fitness_calculator_configuration.py:344]: I think the validation protocols should also be renamed to be consistent

[the remark this answers is not in the document]
Comment (ilkilic) [reply bluepyemodel/evaluation/fitness_calculator_configuration.py:344]: perhaps @darshanmandge can confirm it?

[the remark this answers is not in the document]
Comment (darshanmandge) [reply bluepyemodel/evaluation/fitness_calculator_configuration.py:344]: The user may want to even rename validation protocols. So, I think it is fine to have it is now.

[the remark this answers is not in the document]
Comment (darshanmandge) [reply bluepyemodel/evaluation/fitness_calculator_configuration.py:344]: True.`get_mapped_protocol_name` can be used to get the lower case name for comparison.

**Review by AurelienJaquier — APPROVED** [review]
looks good!

**Pull request merged**